pkgs/build-support/trivial-builders/default.nix

```diff
@@ -347,12 +347,20 @@ rec {
       '';
 
       checkPhase =
+        # GHC (=> shellcheck) isn't supported on some platforms (such as risc-v)
+        # but we still want to use writeShellApplication on those platforms
+        let
+          shellcheckSupported = (builtins.tryEval ghc.bootPkgs.ghc).success;
```

Comment:
Wouldn't it make more sense to `tryEval` shellcheck itself?

Reply:
I tried, but it returns `true`, same for non-boot ghc.  I think it's because the platform support in the ghc expression is implemented with `throw`